src/osdc/Objecter.cc

```diff
@@ -2752,6 +2752,29 @@ int64_t Objecter::get_object_pg_hash_position(int64_t pool, const string& key,
   return p->raw_hash_to_pg(p->hash_key(key, ns));
 }
 
+int Objecter::get_object_osd_position(int64_t pool, const string& name,
+                                          const string& key, const string& ns)
+{
+  shared_lock rl(rwlock);
+
+  int ret;
+  pg_t pg;
+  int up_primary, acting_primary;
+  vector<int> up, acting;
```

Comment:
you do not need, it seems, 'up' or 'acting', and the called functions know how to handle a null pointer to signal unneeded data.

Reply:
This is the case for `OSDMap::pg_to_up_acting_osds` but not `lookup_pg_mapping` as both gets dereferenced.